v2/api/authorization/v1alpha1api20200801preview/role_assignment_spec_arm_types_gen.go

```diff
@@ -36,7 +36,7 @@ type RoleAssignmentProperties_ARM struct {
 	ConditionVersion                   *string                                 `json:"conditionVersion,omitempty"`
 	DelegatedManagedIdentityResourceId *string                                 `json:"delegatedManagedIdentityResourceId,omitempty"`
 	Description                        *string                                 `json:"description,omitempty"`
-	PrincipalId                        *string                                 `json:"principalId,omitempty"`
+	PrincipalId                        *string                                 `json:"principalId,omitempty" optionalConfigMapPair:"PrincipalId"`
```

Comment:
Should these flags be coming through to the generated code?

Reply:
They should, yes. They are used both during code generation and also at runtime by `reflecthelpers.FindOptionalConfigMapReferences`.

The tags are specifically useful because the pairs aren't on their own type, they're intermixed among lots of other properties. They have some name relationships (`X` and `XConfigRef`), but when scanning for them it's much easier to find them with a tag that points out "these are the pairs", versus having to infer it from their names + types.

Reply:
Ahh, I missed that nuance. All good!